### ISSUE TITLE
Is there a way to adjust the obo parsing strictness?

### ISSUE THREAD

**Issue by mwalzer** [althonos/pronto#159]
I wonder if it is possible to successfully parse a obo file that is 'slightly off'.
E.g. UO in its [current version is missing some dbxrefs](https://github.com/bio-ontology-research-group/unit-ontology/issues/44) which pronto, of course, does not like. 
I haven't found a debug level setting on '_schema level_', but that would be helpful in ontology development, too, I suppose.

**Comment by cmungall** [comment]
in this case I think uo.obo is just really problematic and I hope the uo developers fix it. It's hard to write a parser that is lenient for syntax errors. But there are other cases where it would be nice if pronto had a non-strict mode (e.g. dangling classes).

sometimes using the owl file instead will work but in this case pronto doesn't handle OWL individuals (used in an odd UO modeling pattern)

if you install robot you can do this to generate a correct obo file:

`robot convert -I http://purl.obolibrary.org/obo/uo.owl -o /tmp/uo.obo`

I confirmed this does parse

**Comment by matentzn** [comment]
> But there are other cases where it would be nice if pronto had a non-strict mode (e.g. dangling classes; https://github.com/althonos/pronto/issues/156).

I agree! I would also love to see a very basic "repair" comment that can deal with the three most request OBO parsing errors (which will cover 95% of the parsing problems).

**Comment by althonos** [comment]
Yeah, I have the feeling a majority of error is coming from either dangling classes, and broken xrefs. Some other errors (e.g. duplicate labels) are more or less handled silently...

**Comment by cmungall** [comment]
for dangling classes, I don't think a repair is required. This is allowed unless the profile is OBO Basic, or a custom profile that excludes them, seehttps://owlcollab.github.io/oboformat/doc/obo-syntax.html#6.1.2. And these will be the norm for BasePlus files.

we really should be doing a better job with fixing broken xrefs upstream but it's hard. A lot come from CHEBI and maybe with the CHEBI 2.0 infrastructure it will be easier to get these caught pre-release, but for now for practical purposes we have to be more Postelian...

(CHEBI parse currently fails because of "KEGG:C 2339")
